src/github.com/getlantern/balancer/balancer.go

```diff
@@ -79,29 +92,31 @@ func (b *Balancer) Dial(network, addr string) (net.Conn, error) {
 		log.Debugf("Balancer idle for %s, start checking all dialers", idlePeriod)
 		b.checkDialers()
 	}
-	var dialers dialerHeap
 
+	trustedOnly := false
 	_, port, _ := net.SplitHostPort(addr)
-
 	// We try to identify HTTP traffic (as opposed to HTTPS) by port and only
 	// send HTTP traffic to dialers marked as trusted.
 	if port == "" || port == "80" || port == "8080" {
-		if b.trusted.Len() == 0 {
-			return nil, fmt.Errorf("No trusted dialers!")
-		}
-		dialers = b.trusted
-	} else {
-		dialers = b.dialers
+		trustedOnly = true
```

Comment:
Here's the main change. Select from current dialers instead of a copy of dialers.